bolt-sidecar/src/config/commitments.rs

```diff
@@ -0,0 +1,64 @@
+use clap::{ArgGroup, Args};
+use reqwest::Url;
+use serde::Deserialize;
+
+use crate::common::secrets::EcdsaSecretKeyWrapper;
+
+/// Default port for the JSON-RPC server exposed by the sidecar supporting the Commitments API.
+///
+/// 8017 -> BOLT :)
+pub const DEFAULT_RPC_PORT: u16 = 8017;
+
+/// Command-line options for signing constraint messages
+#[derive(Args, Deserialize, Debug, Clone)]
+#[clap(
+    group = ArgGroup::new("commitments-opts").required(true)
+        .args(&["commitments_port", "firewall_rpc"])
+)]
+pub struct CommitmentOpts {
```

Comment:
What do you think of this pattern to match on which method to use?

```env
BOLT_SIDECAR_COMMITMENT_SERVER_PORT=8017
BOLT_SIDECAR_USE_GATEWAYS=true
BOLT_SIDECAR_GATEWAY_URLS="1, 2, 3, ..."
```

```rust
if use_gateways {
	// start the firewall thing
} else {
	// start the server at port
}
```

Reply:
Why adding a "USE" flag when passing the `GATEWAY_URLS` env already conveys that you want to use delegation?
Also, this doesn't simplify the parsing logic of envs:
- what if "USE" is `true` but the list of gateway is empty?
- what if "USE" is unset but both options are provided?

Reply:
responded in the comment above

Reply:
I would use `FIREWALL_RPCS` only, if that's set, disable HTTP server